### ISSUE TITLE
Terraform sdk dependency older than POD RBAC feature on EKS

### ISSUE THREAD

**Issue by tr-fteixeira** [pulumi/pulumi-terraform#523]
**Problem:**
When running pulumi inside an eks cluster, we are not able to leverage the aws eks fine grained capabilities for pod level permissions via service-account introduced on go sdk version 1.23.13
https://aws.amazon.com/blogs/opensource/introducing-fine-grained-iam-roles-service-accounts/

**Reason:**
Currently the dependency version is set to 0.12.9
https://github.com/pulumi/pulumi-terraform/blob/master/go.mod#L8

This terraform version is set to use aws-sdk-go v1.22.0
https://github.com/hashicorp/terraform/blob/v0.12.9/go.mod#L19

**Solution:**
I see the solution being a bump in this version to at least v0.12.11, where it is already updated. 
https://github.com/hashicorp/terraform/blob/v0.12.11/go.mod#L19

**Comment by lukehoban** [comment]
I assume you are ultimately seeing this as an issue related to the `@pulumi/aws` package?  I expect the actual issue is related to the aws-sdk version in https://github.com/pulumi/pulumi-aws.  But a little more detail on what exactly is not working would be helpful (configuring an AWS resource provider, creating an AWS resource, configuring a KMS-based `--secrets-provider`, etc.).

**Comment by tr-fteixeira** [comment]
I believe this is actually on the terraform portion of it, we use terraform and store state files on s3, pulumi is not able to auth with aws to get the state files from those files using the older sdks. 

terraform:state:RemoteStateReference eks  error: Preview failed: error refreshing Terraform state: AccessDenied: Access Denied

altered sample code snippet

```
import * as pulumi from "@pulumi/pulumi";
import * as terraform from "@pulumi/terraform";

const somethingData = 
    new terraform.state.RemoteStateReference("something-service", {
        backendType: "s3",
        bucket: "something-bucket",
        key: "somehting-service.tfstate",
        region: "us-east-2"
    }).getOutput("something_output");
```

From the same pod, i am able to use the aws cli (newer version) to get those files.

**Comment by lukehoban** [comment]
Got it - yes - looks like you are exactly right - thanks for clarification.

**Comment by tr-fteixeira** [comment]
Fixed on #541 
ty
